src/daft-connect/src/display.rs

```diff
@@ -0,0 +1,326 @@
+use std::fmt::Write;
+
+use daft_core::prelude::*;
+
+pub fn to_tree_string(schema: &Schema) -> eyre::Result<String> {
+    let mut output = String::new();
+    // Start with root
+    writeln!(&mut output, "root")?;
+    // Now print each top-level field
+    for (name, field) in &schema.fields {
+        print_field(&mut output, name, &field.dtype, /*nullable*/ true, 1)?;
+    }
+    Ok(output)
+}
```

Comment:
our convention with other text reprs has been to add a `.repr_<format>` method to the struct.

something like this: 

```rs
pub trait SparkDisplay {
  fn repr_spark_string(&self) -> String // or something similarily named
}

impl SparkDisplay for Schema {
  // ...
}

impl SparkDisplay for Field {
  // ...
}

impl SparkDisplay for DataType {
  // ...
}

```